src/visitor.rs

```diff
@@ -616,6 +635,8 @@ impl<'b, 'a: 'b> FmtVisitor<'a> {
             skipped_range: vec![],
             macro_rewrite_failure: false,
             report,
+            base_skip_macro_names,
+            skip_macro_names: Rc::new(RefCell::new(vec![])),
```

Comment:
I think we could initialize `skip_macro_names` with the given `base_skip_macro_names` and remove `base_skip_macro_names` field